packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -160,15 +163,15 @@ class ImageStream extends Diagnosticable {
 
   /// Stop listening for new concrete [ImageInfo] objects and errors from
   /// the `listener`'s associated [ImageErrorListener].
+  ///
+  /// If `listener` has been added multiple times, this removes all instances.
   void removeListener(ImageListener listener) {
     if (_completer != null)
       return _completer.removeListener(listener);
     assert(_listeners != null);
-    for (int i = 0; i < _listeners.length; ++i) {
-      if (_listeners[i].listener == listener) {
+    for (int i = 0; i < _listeners.length; i += 1) {
+      if (_listeners[i].listener == listener)
         _listeners.removeAt(i);
```

Comment:
Would this work if listener is found in the list in sequence, e.g if `_listeners` is `[x, y, listener, listener, z]`, wouldn't we remove the `listener` at index 2 (which puts the second `listener` now at index 2) and then continue to iterate at index 3 leaving us with `[x, y, listener, z]`?

Reply:
with xster's change, this is no longer an issue